packages/react-components/src/AccountName.tsx

```diff
@@ -195,11 +196,49 @@ function AccountName ({ children, className = '', defaultName, label, onClick, o
     }
 
     if (api && isFunction(api.query.identity?.identityOf)) {
-      setName(() =>
-        identity?.display
-          ? extractIdentity(cacheAddr, identity)
-          : extractName(cacheAddr, accountIndex)
-      );
+      if (accountId) {
+        if (nameCache.get(cacheAddr)) {
+          if (nameCache.get(cacheAddr) !== 'notSet') {
+            setName(() =>
+              identity?.display
+                ? extractIdentity(cacheAddr, identity)
+                : extractName(cacheAddr, accountIndex, nameCache.get(cacheAddr))
+            );
+          } else {
+            setName(() =>
+              identity?.display
+                ? extractIdentity(cacheAddr, identity)
+                : extractName(cacheAddr, accountIndex)
+            );
+          }
+        } else {
+          fetch('https://trust.demo.cord.network/api/v1/mapping/getDetails/' + cacheAddr, {
```

Comment:
Can this URL (the 'https://trust.demo.cord.network) be coming from ENV instead of hardcoding?